src/engagement_db_to_coda/engagement_db_to_coda.py

```diff
@@ -109,9 +109,21 @@ def _update_engagement_db_message_from_coda_message(engagement_db, engagement_db
             ws_code = ws_scheme.get_code_with_code_id(label.code_id)
             correct_dataset = coda_config.get_dataset_config_by_ws_code_string_value(ws_code.string_value).engagement_db_dataset
 
+            # Ensure this message isn't being moved to a dataset which it has previously been assigned to.
+            # This is because if the message has already been in this new dataset, there is a chance there is an
+            # infinite loop in the WS labels, which could get very expensive if we end up cycling this message through
+            # the same datasets at high frequency.
+            # If this message has been in this dataset before, crash and wait for this to be manually corrected.
+            # Note that this is a simple but heavy-handed approach to handling what should be a rare edge case.
+            # If we encounter this problem more frequently than expected, upgrade this to a more sophisticated loop
+            # detector/handler.
+            assert correct_dataset not in engagement_db_message.previous_datasets, \
+                f"Engagement db message '{engagement_db_message.message_id}' is being WS-corrected to dataset " \
```

Comment:
```suggestion
                f"Engagement db message '{engagement_db_message.text}, Msg id: {engagement_db_message. message_id}' is being WS-corrected to dataset " \
```
Will be useful to look up in coda for debugging.

Reply:
I can see how that would help here, added, thanks